src/plots/cartesian/layout_defaults.js

```diff
@@ -239,12 +263,60 @@ module.exports = function supplyLayoutDefaults(layoutIn, layoutOut, fullData) {
         handlePositionDefaults(axLayoutIn, axLayoutOut, coerce, {
             letter: axLetter,
             counterAxes: counterAxes[axLetter],
-            overlayableAxes: overlayableAxes,
+            overlayableAxes: getOverlayableAxes(axLetter, axName),
             grid: layoutOut.grid
         });
 
         coerce('title.standoff');
 
+        addMissingMatchedAxis(axLayoutIn.matches);
+
+        axLayoutOut._input = axLayoutIn;
+    }
+
+    // coerce the 'missing' axes
+    i = 0;
+    while(i < missingMatchedAxisIds.length) {
```

Comment:
What about using a `for` loop here instead of `while`?

Reply:
I prefer using `while` when the terminal value (here `missingMatchedAxisIds.length`) can changed inside the loop.

Reply:
Thanks for the note.

Reply:
No problem. I should have highlighted this. Calling `addMissingMatchedAxis` again inside this `while` is how I got the nested cases to work.